lightstep/sdk/metric/number/traits.go

```diff
@@ -0,0 +1,126 @@
+// Copyright The OpenTelemetry Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package number // import "github.com/lightstep/otel-launcher-go/lightstep/sdk/metric/number"
+
+import (
+	"math"
+	"sync/atomic"
+	"unsafe"
+)
+
+// Traits is the generic traits interface for numbers used in the SDK.
+type Traits[N int64 | float64] interface {
+	// FromNumber turns a generic 64bits into the correct machine type.
+	FromNumber(Number) N
```

Comment:
On line 24--yes because `int64 | float64` equals Any. If you meant on line 26 where `Number` is used, no. `Number` is more like "Either", it's 8-bits of data but you have to know its type. In the export path as we leave generic code, the exporter will see a `number.Number` and refer to the `number.Kind` to know which.

Reply:
Apologies, yes i meant line 24 :)